lib/iris/fileformats/cf.py

```diff
@@ -638,8 +644,9 @@ def identify(cls, variables, ignore=None, target=None, warn=True):
                                 warnings.warn(message % (name, nc_var_name))
                         else:
                             # Restrict to only string type.
-                            if np.issubdtype(variables[name].dtype, np.str):
-                                result[name] = CFLabelVariable(name, variables[name])
+                            if _is_str_dtype(variables[name]):
+                                var = variables[name]
```

Comment:
I think maybe you meant to move this assignment outside the `if` to reduce the duplicate indexing?